### PR TITLE
Error reporting

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: src/main/scala/wdl4s/WdlNamespace.scala [hunk 1]
```diff
@@ -106,9 +107,9 @@ case class WdlNamespaceWithWorkflow(importedAs: Option[String],
         (key, tryValue) <- successes
       } yield key -> tryValue.get)
     } else {
-      val errors = failures.values.collect { case f: Failure[_] => f.exception.getMessage }
+      val errors = failures.values.toList.collect { case f: Failure[_] => f.exception }
       // Fine to use .fromListUnsafe because failures is guaranteed to be nonEmpty
```
Comment (ruchim) [inline src/main/scala/wdl4s/WdlNamespace.scala]: Can probably remove this comment now

**Comment by ruchim** [comment]
👍 looks good

File: src/main/scala/wdl4s/exception/LookupException.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import lenthall.exception.ThrowableAggregation
+import wdl4s.GraphNode
+
+sealed trait LookupException { this: Exception => }
+
+/**
+  * When a variable does not reference any known scope in the namespace.
+  */
+case class VariableNotFoundException(variable: String) extends Exception(s"Variable '$variable' not found") with LookupException
```
Comment (geoffjentry) [inline src/main/scala/wdl4s/exception/LookupException.scala]: `final`ize case classes

File: src/main/scala/wdl4s/util/FileUtil.scala [hunk 3]
```diff
@@ -3,7 +3,6 @@ package wdl4s.util
 import scala.util.{Failure, Success, Try}
 
 object FileUtil {
-  // FIXME: Keep?
```
Comment (geoffjentry) [inline src/main/scala/wdl4s/util/FileUtil.scala]: apparently not ;)

**Comment by geoffjentry** [comment]
👍

**Pull request merged**